e2e/Sandbox/MainWindow.xaml.cs

```diff
@@ -219,6 +234,13 @@ private async void CreateDashboardWithTypeBtn_Click(object sender, RoutedEventAr
             if (creator != null)
             {
                 RdashDocument document = creator.CreateDashboard();
+
+                if(document == null)

```

Comment:
instead of retuning null and showing a messagebox, update the datasource creators to throw exceptions instead. then you can get rid of this code. No need for try/catch either. Just let it crash.

Reply:
updated